plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1415,7 +1415,7 @@ producer_plugin_impl::start_block_result producer_plugin_impl::start_block() {
 
    const auto& hbs = chain.head_block_state();
 
-   if( chain.get_terminate_at_block() > 0 && chain.get_terminate_at_block() < hbs->block_num ) {
+   if( ( chain.get_terminate_at_block() > 0 && chain.get_terminate_at_block() < hbs->block_num ) || (chain.get_config().read_only && chain.get_config().read_mode == db_read_mode::IRREVERSIBLE && chain.last_irreversible_block_num() < hbs->block_num )) {
```

Comment:
How is it possible that `chain.last_irreversible_block_num() < hbs->block_num` is ever true?
I can't see how this fixes the issue.
Where you able to reproduce the reported issue?

Reply:
Can't repro using the way you told me, I wrote the steps under EPE915 and @heifner , but no response from you.

Reply:
Today reproduced, and change the code , and test it works. and then commit again.